app/src/main/java/com/ncnf/map/MapHandler.java

```diff
@@ -0,0 +1,244 @@
+                    if (e instanceof ApiException){
+                        ApiException apiException = (ApiException) e;
+                        apiException.printStackTrace();
+                        int statusCode = apiException.getStatusCode();
+                        Log.i("Place fetching", "Place not found: " + e.getMessage());
+                        Log.i("Place fetching", "Status code: " + statusCode);
+                    }
+                });
+            }
+
+            @Override
+            public void OnItemDeleteListener(int position, View v) {
+
+            }
+        });
+    }
+
+    public void switchMarkers(GoogleMap mMap) {
+        eventsShown = !eventsShown;
+        for (Marker e : eventMarkers) {
+            e.setVisible(eventsShown);
+        }
+        for (Marker v : venueMarkers) {
+            v.setVisible(!eventsShown);
+        }
+        if (eventsShown) {
+            mMap.setContentDescription("MAP_WITH_EVENTS");
+        } else {
+            mMap.setContentDescription("MAP_WITH_VENUES");
+        }
+    }
+
+    public void show_markers(GoogleMap mMap, EventProvider eventProvider, VenueProvider venueProvider){
+        // Add a marker near EPFL and move the camera
+        MarkerOptions position_marker = new MarkerOptions().position(userPosition).title("Your Position").icon(Utilities.bitmapDescriptorFromVector(context));
+        userMarker = mMap.addMarker(position_marker);
+        mMap.moveCamera(CameraUpdateFactory.newLatLng(userPosition));
+        mMap.moveCamera(CameraUpdateFactory.zoomTo(zoomLevel));
+
+        // Add a marker for each event
+        List<Event> events = eventProvider.getAll();
+        eventMarkers = new ArrayList<>();
+        for (Event p : events) {
+            LatLng event_position = new LatLng(p.getLatitude(), p.getLongitude());
+            if (Utilities.position_in_range(event_position, userPosition)){
+                eventMarkers.add(mMap.addMarker(new MarkerOptions().position(event_position).title(p.getName())));
+            }
+        }
+        // Add a marker for each venue
+        List<Venue> venues = venueProvider.getAll();
+        venueMarkers = new ArrayList<>();
+        for (Venue p : venues) {
+            LatLng venue_position = new LatLng(p.getLatitude(), p.getLongitude());
+            if (Utilities.position_in_range(venue_position, userPosition)){
+                venueMarkers.add(mMap.addMarker(new MarkerOptions().position(new LatLng(p.getLatitude(), p.getLongitude())).title(p.getName())));
+                //Set last element (above line) to invisible
+                venueMarkers.get(venueMarkers.size() - 1).setVisible(false);
+            }
+        }
+    }
+
+    public void setZoomLevel(float zoomLevel) {
+        this.zoomLevel = zoomLevel;
+    }
+
+    //Removes all markers from the map and recreates them according to current position
+    public void update_markers(){
+        for (Marker m : eventMarkers)
```

Comment:
single for fori loop?

Reply:
Not the same range, eventMarkers and venueMarkers do not have the same size